File: android/src/main/java/top/huic/tencent_im_plugin/message/entity/LocationMessageEntity.java
package top.huic.tencent_im_plugin.message.entity;

import com.tencent.imsdk.v2.V2TIMLocationElem;

import top.huic.tencent_im_plugin.enums.MessageNodeType;

/**
 * 位置消息实体
 *
 * @author 蒋具宏
 */
public class LocationMessageEntity extends AbstractMessageEntity {
    /**
     * 描述
     */
    private String desc;

    /**
     * 经度
     */
    private double latitude;

    /**
     * 纬度
     */
    private double longitude;

    public LocationMessageEntity() {
        super(MessageNodeType.Location);
    }

    public LocationMessageEntity(V2TIMLocationElem elem) {
        super(MessageNodeType.Location);
        this.setDesc(elem.getDesc());
        this.setLongitude(elem.getLongitude());
        this.setLatitude(elem.getLatitude());
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }
}

File: ios/Classes/message/entity/LocationMessageEntity.swift
import ImSDK

//  Created by 蒋具宏 on 2020/3/15.
//  位置消息实体
public class LocationMessageEntity: AbstractMessageEntity {

    /// 描述
    var desc: String?;

    /// 经度
    var latitude: Double?;

    /// 纬度
    var longitude: Double?;

    override init() {
        super.init(MessageNodeType.Location)
    }

    init(elem: V2TIMLocationElem) {
        super.init(MessageNodeType.Location)
        self.desc = elem.desc;
        self.longitude = elem.longitude;
        self.latitude = elem.latitude;
    }
}
